src/test/java/WebDriverScenarioWithTests.java

```diff
@@ -0,0 +1,122 @@
+                .elementToBeClickable(By.xpath("//input[@name='btnK']"))).click();
+        List<WebElement> searchResults = driver.findElements(By.xpath("//cite[contains(text(),'wikipedia')]"));
+        logger.info(String.format("Search results number for requested text: %s", searchResults.size()));
+
+        //3. Validation
+        WebElement firstElementOfSearchResults = driver.findElement(By.xpath("//cite[@class='iUh30 Zu0yb qLRx3b tjvcx']"));
+        String contentOfFirstSearchResult = firstElementOfSearchResults.getText();
+        new WebDriverWait(driver, 50).until(ExpectedConditions
+                .presenceOfAllElementsLocatedBy(By.xpath("//cite[contains(text(),'wikipedia')]")));
+        logger.info(String.format("First element of search results is %s", contentOfFirstSearchResult));
+            wait.until(pageLoadCondition);
+        Assert.assertTrue(contentOfFirstSearchResult.contains("wikipedia"), "First element of search results do not contain expected text.");
+
+        //4. Navigate to https://en.wikipedia.org via first found link
+        WebElement firstEnglishWikipediaLink = driver.findElement(By.xpath("//cite[text()='https://en.wikipedia.org']"));
+        new WebDriverWait(driver, 20).until(ExpectedConditions
+                .presenceOfAllElementsLocatedBy(By.xpath("//cite[text()='https://en.wikipedia.org']")));
+        String contentOfFirstEnglishWikipediaLink = firstEnglishWikipediaLink.getText();
+        firstEnglishWikipediaLink.click();
+            wait.until(pageLoadCondition);
+
+        //5. Assert that we are currently on wikipedia page;
+        logger.info(String.format("The link clicked is %s", contentOfFirstEnglishWikipediaLink));
+
+        Assert.assertTrue(contentOfFirstEnglishWikipediaLink.contains("wikipedia"), "The element is clicked do not contains expected text.");
+
+        //6. Get Page Title name and Title length -> Print Page Title and Title length on the Console via logger;
+        String titleOfClickedEnglishWikipediaLink = driver.getTitle();
+        int titleLength = titleOfClickedEnglishWikipediaLink.length();
+        logger.info("The title of clicked link is " + titleOfClickedEnglishWikipediaLink
+                + ".\n Title length is " + titleLength + " symbols.");
+
+        //7. Get Page URL and verify if the it is a correct page opened;
+        String clickedURL = driver.getCurrentUrl();
+
+        Assert.assertTrue(clickedURL.contains("wikipedia"), "The opened link do not content expected text.");
+
+        //8. Print Page Length on Console via logger;
+        long clickedURLLength = clickedURL.length();
+        logger.info("The full page URL is clicked contains " + clickedURLLength + " symbols.");
+
+        //9. Take Screenshot of the first Image  in the “Did you know...” container
+
+            wait.until(pageLoadCondition);
+
+        WebElement didYouKnowImage = driver.findElement(By.xpath(".//div[contains(@id,'dyk')]//img"));
```

Comment:
Use first found image, because it is possible, that DYK can contain >1 image in it block

Reply:
As we saw, that on Russian wikipedia site we have 3 pictures, recommends to work with first found one